include/cutlass/gemm/collective/xe_mma.hpp

```diff
@@ -210,31 +234,31 @@ struct CollectiveMma<
     (void)residue_mnk;
     (void)thread_idx;
     (void)smem_buf;
-
-    auto tiled_copy_a = make_xe_2d_copy(atom_load_A{}.with(mainloop.mA),
-                                             Layout<Shape<_1, Int<SubgroupSize>>>{});
-    auto tiled_copy_b = make_xe_2d_copy(atom_load_B{}.with(mainloop.mB),
-                                             Layout<Shape<_1, Int<SubgroupSize>>>{});
-
-    // Partition the copying of A and B tiles across the threads
-    auto thr_copy_A = tiled_copy_a.get_slice(thread_idx);
-    auto thr_copy_B = tiled_copy_b.get_slice(thread_idx);
+    
+    auto thr_copy_A2 = mainloop.copy_A.get_slice(thread_idx);
+    auto thr_copy_B2 = mainloop.copy_B.get_slice(thread_idx);
 
     // Instantiate the MMA object and get thread slice
     TiledMma tiled_mma;
-    auto thr_mma = tiled_mma.get_slice(thread_idx);
+    // To make all threads in a warp have the same global tensors pass in the index of thread 0 in each warp
```

Comment:
Can we have a `TODO(Codeplay):` here to fix this later?

Reply:
done